src/compiler/checker.ts

```diff
@@ -5366,6 +5366,7 @@ namespace ts {
 
                 if (expectsIdentifier && chain.length !== 1
                     && !context.encounteredError
+                    && !(context.flags & NodeBuilderFlags.AllowQualifiedNameInPlaceOfIdentifier)
                     && !(context.flags & NodeBuilderFlags.AllowQualifedNameInPlaceOfIdentifier)) {

```

Comment:
the new value is the same as the old, so you should replace the old usage instead of adding to it.

Reply:
Fixed